### PR TITLE
[deploy] Crawl for dynamic library dependencies and copy them to the roborio

### PR DESCRIPTION
**Pull request by pjreiniger**
Leaves Bazel at 4.2.1, but gives example how to crawl for dynamic libs need to copy over.

Since I don't have a roborio, it just prints out that the files do in fact exist.

### CODE REVIEW

**Comment by connorworley** [comment]
Everything here makes sense - I'll try to make deploy.py copy things over as a next step. Maybe this is also where #33 comes in.

File: bazelrio/defs.bzl [hunk 1]
```diff
@@ -36,11 +58,19 @@ def robot_cc_binary(name, team_number, srcs = [], hdrs = [], deps = [], halsim_c
...
+    )
+
+    base_deploy_command = "$(location @bazelrio//scripts/deploy) --robot_binary $(location {}) --team_number {} --dynamic_libraries $(locations {})".format(name, team_number, discover_dynamic_deps_task_name)
     native.genrule(
         name = "{}.deploy".format(name),
-        srcs = [":{}".format(name)],
+        srcs = [":{}".format(name), discover_dynamic_deps_task_name],
         tools = ["@bazelrio//scripts/deploy"],
-        outs = ["deploy"],
+        outs = ["deploy.bat"],
```
[the remark this answers is not in the document]
Comment (noamzaks) [reply bazelrio/defs.bzl:72]: Oy

**Comment by connorworley** [comment]
@pjreiniger what's the next step from here? Still want me to update the deploy script?

**Comment by pjreiniger** [comment]
Oh whoops, I didn't realize I made this from my fork. I just cleaned it out last night

**Comment by connorworley** [comment]
Adding a naive 
```
   for dylib_path in args.dynamic_libraries:
        dylib_name = basename(dylib_path)
        sftp_client.put(dylib_path, f"/home/lvuser/{dylib_name}")
```
does seem to work with `LD_LIBRARY_PATH=/home/lvuser` prepended to the robotCommand.

**Comment by connorworley** [comment]
Probably better to just upload straight to `/usr/local/frc/third-party/`

File: bazelrio/scripts/deploy/deploy.py [hunk 2]
```diff
@@ -74,6 +75,9 @@ def deploy(argv):
     sftp_client.chmod("/home/lvuser/robotCommand", 0o755)
     sftp_client.chown(destination_path, 500, 500)
     sftp_client.chown("/home/lvuser/robotCommand", 500, 500)
+    for dylib_path in args.dynamic_libraries:
+        dylib_name = basename(dylib_path)
+        sftp_client.put(dylib_path, f"/usr/local/frc/third-party//{dylib_name}")
```
Comment (noamzaks) [inline bazelrio/scripts/deploy/deploy.py:80]: Isn't there an extra /?

File: bazelrio/defs.bzl (hunk 1 above)
[the remark this answers is not in the document]
Comment (noamzaks) [reply bazelrio/defs.bzl:72]: Maybe you can use the script_suffix here also?

**Review by noamzaks — APPROVED** [review]

File: bazelrio/defs.bzl (hunk 1 above)
[the remark this answers is not in the document]
Comment (connorworley) [reply bazelrio/defs.bzl:72]: alternatively, we could just use .bat everywhere and remove the script_suffix idea from the rio toolchain

**Pull request merged**